enhancements/olm-toggle-copied-csvs.md

```diff
@@ -0,0 +1,199 @@
+
+#### Orphaned CSVs
+
+Orphaned CSVs are CSVs that were installed via a `Subscription` that no longer exists. Since the lifetimes -- i.e. the period of existence on a cluster --  of `Operator` resources aren't directly tied to the lifetimes of the resources that triggered their generation, the `Operator` resource(s) associated with Orphaned CSVs will continue to exist _after_ a `Subscription` is deleted. If that `Operator` resource is deleted explicitly, OLM will generate another as per the rules [outlined above](#operator-cardinality).
+
+#### Copied RBAC 
+
+In addition to operator discoverability, Copied CSVs also serve as convienient vehicles for the garbage collection of Copied RBAC. These are resources that are copied along with the CSV to provide the operator access to a namespace. To support the loss of Copied CSVs, the RBAC copying logic in OLM will be extended to write additional owner references from copied resources to the related Operator resource. 
+
+#### The Toggle
+
+A feature toggle will be added to control OLM's use of Copied CSVs. When enabled (toggled on), OLM will delete all existing Copied CSVs and will not generate any more. When disabled (toggled off), OLM will generate Copied CSVs (much like it does today).
+
+When toggled (on then off), OLM will recreate any missing Copied CSVs and reassociate the owner references of related copied resources with them (i.e. the inverse operation of the one described in the [Copied RBAC](#copied-rbac) section).
+
+The toggle will be disabled by default.
+
+The toggle can be set from a command-line option on the olm-operator binary.
+
+#### Config CRD w/ Toggle
+
+A novel cluster-scoped `OLMConfig` CRD will be added to OLM. The resource type it defines will allow users to apply a limited set of configurations to the OLM instance on their cluster by creating or modifying a singlton (of that type) on the cluster.
+
+The name of that singleton will be `config` and OLM will ignore all other `OLMConfig` resources on a cluster.
+
+Its spec will contain a Copied CSV feature toggle field.
+
+e.g.
+
+```yaml
+apiVersion: operators.coreos.com/v1
+kind: OLMConfig
+metadata:
+  name: config
+spec:
+  features:
+    'disable-copied-csvs': true
+```
+
+When both the [command-line option](#the-toggle) _and_ `config` toggle field are present, the latter takes precedence.
+
+When the `config` resource isn't present, OLM uses its default settings.
+
+### Risks and Mitigations
+
+#### Loss of Operator Discovery
+
+**Risks:** When Copied CSVs are disabled, users won't be able to discover what operators are configured to watch a particular namespace unless they have access to their install namespaces.
+
+**Potential Mitigations:** 
+
+- Reduce number of impacted users by limiting toggle to operators installed in `AllNamespace` mode only; that is, `Single` and `MultiNamespace` mode Copied CSVs will not be affected by the toggle.
```

Comment:
is this better, or will it be more confusing the user that they see some, but not all, of the installed operators?

Reply:
I think it will be **more confusing**, but I've received feedback in other venues along the lines of "the confusion is worth the benefit of retaining discovery for users of Single/MultiNamespace operators"

I thought this was reasonable, which is why I changed the EP.

Reply:
ok.  eventually having the console provide some indication that this is the situation would definitely be useful.